diff -Nurb /home/br1/wpa_supplicant-0.3.9/driver_madwifi.c Madwifi-20050910/wpa/wpa_supplicant-0.3.9/driver_madwifi.c
--- /home/br1/wpa_supplicant-0.3.9/driver_madwifi.c	2005-05-19 05:20:27.000000000 +0200
+++ Madwifi-20050910/wpa/wpa_supplicant-0.3.9/driver_madwifi.c	2005-08-20 20:43:20.000000000 +0200
@@ -19,6 +19,7 @@
 #include <string.h>
 #include <sys/ioctl.h>
 #include <errno.h>
+#include <sys/socket.h>
 
 #include "common.h"
 #include "driver.h"
@@ -76,21 +77,24 @@
 				"ioctl[IEEE80211_IOCTL_SETPARAM]",
 				"ioctl[IEEE80211_IOCTL_GETPARAM]",
 				"ioctl[IEEE80211_IOCTL_SETKEY]",
-				"ioctl[IEEE80211_IOCTL_GETKEY]",
+				"ioctl[IEEE80211_IOCTL_SETWMMPARAMS]",
 				"ioctl[IEEE80211_IOCTL_DELKEY]",
-				NULL,
+				"ioctl[IEEE80211_IOCTL_GETWMMPARAMS]",
 				"ioctl[IEEE80211_IOCTL_SETMLME]",
-				NULL,
+				"ioctl[IEEE80211_IOCTL_GETCHANINFO]",
 				"ioctl[IEEE80211_IOCTL_SETOPTIE]",
 				"ioctl[IEEE80211_IOCTL_GETOPTIE]",
 				"ioctl[IEEE80211_IOCTL_ADDMAC]",
 				NULL,
 				"ioctl[IEEE80211_IOCTL_DELMAC]",
+				"ioctl[IEEE80211_IOCTL_GETCHANLIST]",
+				"ioctl[IEEE80211_IOCTL_SETCHANLIST]",
 				NULL,
-				"ioctl[IEEE80211_IOCTL_CHANLIST]",
+				"ioctl[IEEE80211_IOCTL_SETMODE]",
+				"ioctl[IEEE80211_IOCTL_GETMODE]",
 			};
 			if (IEEE80211_IOCTL_SETPARAM <= op &&
-			    op <= IEEE80211_IOCTL_CHANLIST)
+			    op <= IEEE80211_IOCTL_GETMODE)
 				perror(opnames[op - SIOCIWFIRSTPRIV]);
 			else
 				perror("ioctl[unknown???]");
